### PR TITLE
Initial host config, nixos setup for ghaf-proxy

### PR DESCRIPTION
**Pull request by fayadf**
Initial server configuration for ghaf-proxy server, and initial nixos deployment on top of linux distro. Changes include

1. Configuration for the server
2. Disk formatting and mounts
3. Including the new server into existing deployment
4. Added existing team into server for ssh

### CODE REVIEW

**Review by joinemm — CHANGES_REQUESTED** [review]

**Comment by karim20230** [comment]
@fayadf Missing copyright/licensing information

File: hosts/default.nix [hunk 1]
```diff
@@ -83,6 +84,7 @@ in
         "testagent-dev"
         "ghaf-log"
         "ghaf-coverity"
+	      "ghaf-proxy"
```
[the remark this answers is not in the document]
Comment (fayadf) [reply hosts/default.nix]: Fixed indentation

File: hosts/ghaf-proxy/configuration.nix [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      user-avnik
+      user-bmg
+      user-flokli
+      user-hrosten
+      user-ktu
+      user-mariia
+      user-mkaapu
+      user-vilvo
+      user-vjuntunen
+      user-vunnyso
```
[the remark this answers is not in the document]
Comment (fayadf) [reply hosts/ghaf-proxy/configuration.nix]: Removed users

**Comment by fayadf** [comment]
@karim20230 Ack! Fixed it.

**Pull request closed without merging**